lib/active_record/typed_store/dsl.rb

```
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
   class DSL
     attr_reader :fields, :coder
 
-    def initialize(attribute_name, options)
-      @coder = options.fetch(:coder) { default_coder(attribute_name) }
-      @accessors = options[:accessors]
-      @accessors = [] if options[:accessors] == false
+    def initialize(store_name, options)
+      @coder = options.fetch(:coder) { default_coder(store_name) }
+      @store_name = store_name
+      @prefix =
+        case options[:prefix]
+        when String, Symbol
+          "#{options[:prefix]}_"
+        when TrueClass
+          "#{store_name}_"
+        else
+          ""
+        end
+      @suffix =
+        case options[:suffix]
+        when String, Symbol
+          "_#{options[:suffix]}"
+        when TrueClass
+          "_#{store_name}"
+        else
```

Same here. Let's raise if an unexpected type is passed.